source/docs/advanced/theme/route-dispatcher.md

```diff
@@ -0,0 +1,134 @@
+---
+id: route-dispatcher
+title: Handle dynamic URLs with the Dispatcher
+---
+
+In some cases, you will need a deeper customization of URLs than what you can do in the [Add a new page guide](../../essentials/add-a-page-client-side). For instance, for some SEO reasons, you may prefer to have `/my-product` instead of `/product/my-product-slug`.
+
+That is what the Dispatcher is for in Front-Commerce, and what we will cover in this documentation.
+
+## What is the goal of the Dispatcher?
+
+The dispatcher is actually a component within Front-Commerce that will be displayed in case no other route was found. Its goal will be to ask the server (by using the `matchUrls` query in your GraphQL Schema) what kind of page is associated with the current URL and will display the page's component accordingly.
+
+<figure>
```

Comment:
It would be great to have a sentence here to introduce the schema.

Proposal: « Below is a flowchart illustrating the URL resolution logic »